File: MythonC++/Mython/Type.cpp
#include "Type.h"

Type::Type()
{
	_isTemp = false;
}

Type::Type(bool isTemp)
{
	_isTemp = isTemp;
}

bool Type::IsTemp()
{
	return _isTemp;
}

void Type::SetTemp(bool isTemp)
{
	_isTemp = isTemp;
}

File: MythonJava/src/sample/types/Type.java
package sample.types;

public abstract class Type {

    protected boolean isTemp;

    public Type() {
        this(false);
    }

    public Type(boolean isTemp) {
        this.isTemp = isTemp;
    }

    public void setTemp(boolean temp) {
        this.isTemp = temp;
    }

    public abstract boolean isPrintable();

    public abstract String toString();

    public abstract String getType();

}
